### DISCUSSION TITLE
Play wav simultaneously + asynchronously on PWM with ESP32

### DISCUSSION THREAD

**Discussion by louis-badr** [pschatzmann/arduino-audio-tools#1047] [Q&A] [answered]
I generated some files and managed to read them on an ESP32 with PWM.
Now I'd like to be able to listen to the serial port to read files on demand, i.e. to be able to choose which file to play, when to start and stop it, on which pin to read it, at what volume and have them be able to overlap.
What I've tried doesn't work, is this possible at all?
Thanks for the library by the way! 😄

**Comment by pschatzmann** [comment, marked as answer]
I am sorry, I don't quite understand what you are doing and what you want to achieve: files and PWM ? 

But I have the gut feeling that you might want to read the chapter about mixing...

**Reply by louis-badr** [reply]
Sorry if that wasn't clear.
I have several wav files converted to C arrays and several speakers connected to my ESP32.
I would like to play one of the files on one of the speakers with PWM and for it to be possible to play another file on another speaker at any time without interrupting the first one or waiting for it to finish.
Is this possible with mixing?

**Comment by pschatzmann** [comment]
I think you can not have multiple instances of  PWMAudioOutput. But one instance supports __many output pins (one pin for each channel)__. So you will need to write some logic to combine your different files into one output where each file goes into a separate channel. 

The [InputMerge](https://github.com/pschatzmann/arduino-audio-tools/wiki/Splitting-and-Merging-Audio) should help to to do this if your files are mono. If they are stereo, you will  need to [convert them to mono](https://github.com/pschatzmann/arduino-audio-tools/wiki/Converting-the-Data-Format) first. Or you can just implement your own logic using [reads from your input and writes](https://github.com/pschatzmann/arduino-audio-tools/wiki/Introduction)... 

Mixing is about playing your different input files at the same time on 1 speaker (for mono) or on 2 speakers (for stereo)

**Reply by louis-badr** [reply]
Ok thanks, I'll try to use InputMerge.
Other than that, is it possible to use a single channel with PWMAudioOutput but reassign the output pin?
I've tried changing it with `config.start_pin = ...;` or `config.setPins(...);`, or creating two PWMConfig but without success. After the sound has been played once on a pin, even if the variable has been modified, I can't seem to play it on another pin.
